frontend/scripts/react-components/tool/tool.selectors.js

```diff
@@ -1,37 +1,61 @@
 import { createSelector } from 'reselect';
 import { getSelectedContext } from 'reducers/app.selectors';
+import { MIN_COLUMNS_NUMBER } from 'constants';
 
 const getToolSelectedNodesIds = state => state.toolLinks.selectedNodesIds;
 const getToolNodes = state => state.toolLinks.data.nodes;
 const getToolColumns = state => state.toolLinks.data.columns;
 const getToolSelectedColumnsIds = state => state.toolLinks.selectedColumnsIds;
+const getExtraColumnId = state => state.toolLinks.extraColumnId;
 const getHighlightedNodeIds = state => state.toolLinks.highlightedNodeId;
 
 export const getSelectedColumnsIds = createSelector(
-  [getSelectedContext, getToolColumns, getToolSelectedColumnsIds],
-  (selectedContext, columns, selectedColumnsIds) => {
+  [getSelectedContext, getToolColumns, getToolSelectedColumnsIds, getExtraColumnId],
+  (selectedContext, columns, selectedColumnsIds, extraColumnId) => {
     if (
       selectedColumnsIds &&
       selectedContext &&
-      selectedColumnsIds.length === selectedContext.defaultColumns.length
+      selectedColumnsIds.length === selectedContext.defaultColumns.length &&
+      !extraColumnId
     ) {
       return selectedColumnsIds;
     }
 
     if (!columns && !selectedContext) {
       return [];
     }
-    return selectedContext.defaultColumns.reduce((acc, column) => {
+
+    const selectedColumns = selectedContext.defaultColumns.reduce((acc, column) => {
       let id = column.id;
       if (selectedColumnsIds && selectedColumnsIds[column.group]) {
         id = selectedColumnsIds[column.group];
       }
-      acc[column.group] = id;
+      acc.push(id);
+
+      if (columns && columns[id].filterTo && columns[id].filterTo === extraColumnId) {
```

Comment:
this is the reason why we cant load links and columns in parallel, also the reason why we have race conditions and why we need to include the toolColumns everywhere. What if we also persist the parent column?